README.md

```diff
@@ -1,6 +1,149 @@
 # Keycloak-deployment
 
-
 This repository contains all our scripts to deploy keycloak on Openshift and minishift.
 Also we have scripts to bake our own docker image using the keycloak source code
 from our repository `almighty/keycloak`.
+
+# Almighty-Keycloak Docker Image
+
+To build this image it is necessary to have previously generated the executables of this
+project. To do this, run the following Maven command in the almighty/keycloak repository:
+
+`$ mvn clean install -DskipTests -pl :keycloak-server-dist -am -P distribution`
+
+This generates some tarballs with the required executables. To build the docker image,
+copy the generated tar file (e.g. `keycloak-3.0.0.Final.tar.gz`) from the almighty/keycloak
+repository into the docker folder, like so:
+
+`$ cp $KEYCLOAK_REPO/distribution/server-dist/target/keycloak-3.0.0.Final.tar.gz $KEYCLOAK_DEPLOYMENT_REPO/docker`
+
+Then you just need to build the docker image.  Change into the docker directory and run the following command:
+
+`$ docker build --tag IMAGE_NAME .`
+
+If you would like to build image for clustered mode add build argument
+
+`$ docker build  --build-arg OPERATING_MODE=clustered --tag IMAGE_NAME .`
+
+Note that, this docker image installs the certificate to securely talk to OpenShift Online.
```

Comment:
I still needed to do it to test oAuth between KC and minishift, before installing the certificate my log was full of certificate-related errors and nothing worked.